pkg/trace/mapper_trace_parser.go

```diff
@@ -0,0 +1,109 @@
+package trace
+
+import (
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"os"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/vhive-serverless/loader/pkg/common"
+)
+
+type MapperTraceParser struct {
+	DirectoryPath         string
+	duration              int
+	functionNameGenerator *rand.Rand
+}
+
+type DeploymentInfo struct {
+	YamlLocation          string
+	PredeploymentCommands []string
+}
+
+type JSONParser map[string]DeploymentInfo
+
+func NewMapperParser(directoryPath string, totalDuration int) *MapperTraceParser {
+	return &MapperTraceParser{
+		DirectoryPath: directoryPath,
+
+		duration:              totalDuration,
+		functionNameGenerator: rand.New(rand.NewSource(time.Now().UnixNano())),
+	}
+}
+
+func (p *MapperTraceParser) extractFunctions(mapperOutput map[string]map[string]string, deploymentInfo JSONParser, dirPath string) []*common.Function {
+	var result []*common.Function
+	invocations := parseInvocationTrace(dirPath+"/invocations.csv", p.duration)
+	runtime := parseRuntimeTrace(dirPath + "/durations.csv")
+	memory := parseMemoryTrace(dirPath + "/memory.csv")
+
+	runtimeByHashFunction := createRuntimeMap(runtime)
+	memoryByHashFunction := createMemoryMap(memory)
+	for i := 0; i < len(*invocations); i++ {
+		invocationStats := (*invocations)[i]
+		hashFunction := invocationStats.HashFunction
+		proxyFunction := mapperOutput[hashFunction]["proxy-function"]
+		yamlPath := deploymentInfo[proxyFunction].YamlLocation
+
+		function := &common.Function{
+			Name: fmt.Sprintf("%s-%d-%d", proxyFunction, i, p.functionNameGenerator.Uint64()),
+
+			InvocationStats: &invocationStats,
+			RuntimeStats:    runtimeByHashFunction[hashFunction],
+			MemoryStats:     memoryByHashFunction[hashFunction],
+			YAMLPath:        yamlPath,
+		}
+		result = append(result, function)
+	}
+	return result
+}
+func (p *MapperTraceParser) Parse() []*common.Function {
+	var functions []*common.Function
```

Comment:
Add spaces before for, return and func.

Reply:
Add spaces before for, return and func? `gomft` seems to be giving out no diffs, do you mean blank lines?